src/System.Management.Automation/engine/InitialSessionState.cs

```diff
@@ -4495,14 +4495,37 @@ static InitialSessionState()
             builtinVariables.Add(
                 new SessionStateVariableEntry(
                     SpecialVariables.NativeArgumentPassing,
-                    Platform.IsWindows ? NativeArgumentPassingStyle.Windows : NativeArgumentPassingStyle.Standard,
+                    GetPassingStyle(), // Platform.IsWindows ? NativeArgumentPassingStyle.Windows : NativeArgumentPassingStyle.Standard,
                     RunspaceInit.NativeCommandArgumentPassingDescription,
                     ScopedItemOptions.None,
                     new ArgumentTypeConverterAttribute(typeof(NativeArgumentPassingStyle))));
 
             BuiltInVariables = builtinVariables.ToArray();
         }
 
+        /// <summary>
+        /// Assigns the default behavior for native argument passing.
+        /// If the system is non-Windows, we will return Standard.
+        /// If the PSVersion takes the shape of a Version Type, we will return Legacy.
+        /// This is because our preview and daily builds have a PSVersion which
+        /// looks like 7.3.0-preview.3 which can't be parsed into a Version.
```

Comment:
This needs to be updated, maybe just say "Otherwise, return Legacy for stable release and Windows for preview release".

Reply:
updated